packages/cli/src/models/dependency/Dependency.ts

```diff
@@ -89,7 +89,7 @@ export default class Dependency {
     // to Projects, which handle library deployment and linking for a set of contracts altogether.
 
     const contracts = this.projectFile.contracts.map((contractName): [Contract, string] => [
-      Contracts.getFromNodeModules(this.name, contractName),
+      Contracts.getFromNodeModules(this.name, contractName).upgradeable,
```

Comment:
It's hard to say that because nothing prevents using this class at an earlier stage. We would have to model things differently if we want to be able to ensure that at compile time.

Reply:
Yes, that is the assumption in this code.